File: 14-longest-common-prefix/14-longest-common-prefix.java
class Solution {
    public String longestCommonPrefix(String[] strs) {
        Arrays.sort(strs);//do check this on java visualizer
        String start=strs[0];
        String end=strs[strs.length-1];
        int i=0;
        while(i<start.length())//note that length of start is considered and not end
        {
            if(start.charAt(i)==end.charAt(i)) 
                i++;
            else 
                break;
        }
        if(i==0){
            return "";
        }else{
            return start.substring(0,i);
        }  
    }
}

File: 14-longest-common-prefix/14-longest-common-prefix.cpp
class Solution {
public:
    string longestCommonPrefix(vector<string>& strs) {
        sort(strs.begin(), strs.end());

        string temp1 = strs[0], temp2 = strs[strs.size()-1];
        int n = temp1.size();
        int i=0; 
        string ans = "";
        while(i<n){
            if(temp1[i] != temp2[i])
                break;
            else
                ans += temp1[i];
                i++;
        }
        return ans;
    }
};